cypress/e2e/signIn.cy.js

```diff
@@ -12,23 +17,54 @@ describe('Sign In page', () => {
 
   before(() => {
     cy.task('db:clear');
+  });
+
+  beforeEach(() => {
     cy.task('generateUser').then((generateUser) => {
       user = generateUser;
+      signInPage.visit();
+      cy.registerUser(user.email, user.username, user.password);
     });
   });
 
   it('should provide an ability to log in with existing credentials', () => {
-    signInPage.visit();
-    cy.register(user.email, user.username, user.password);
-
-    signInPage.typeEmail(user.email);
-    signInPage.typePassword(user.password);
-    signInPage.clickSignInBtn();
+    signInPage.enterLoginCredentials(user.email, user.password);
+    signInPage.clickOnSignInButton();
```

Comment:
The `clickOnSignInButton` method is not defined in the SignInPageObject class. Make sure you have it properly defined and imported.

Reply:
![e2e_15](https://github.com/mate-academy/qa_cypress_e2e_project/assets/13405550/82dd01a0-5e6c-4e8a-893b-b0cc40f46eba)